dex-it/src/test/scala/com/wavesplatform/it/sync/api/ws/WsInternalStreamTestSuite.scala

```diff
@@ -0,0 +1,431 @@
+        buffer should have size 1
+
+        val orderEvents = buffer.orderEvents
+        orderEvents.keys should have size 1
+        orderEvents.keys.head shouldBe order.id()
+
+        orderEvents(order.id()) should matchTo {
+          List(
+            mkCancelledFullOrder(
+              order,
+              filledAmount = 0,
+              filledFee = 0,
+              avgWeighedPrice = 0,
+              isMarket = false
+            ))
+        }
+
+        wsc.close()
+      }
+
+      "multiple matches" in {
+        val order1 = mkOrderDP(bob, wavesUsdPair, OrderType.SELL, 2.waves, 3, matcherFee = 0.004.btc, feeAsset = btc)
+        val order2 = mkOrderDP(bob, wavesUsdPair, OrderType.SELL, 2.waves, 2, matcherFee = 0.003.waves, feeAsset = Waves)
+        val order3 = mkOrderDP(alice, wavesUsdPair, OrderType.BUY, 4.waves, 4, matcherFee = 0.003.waves, feeAsset = Waves)
+        val orders = List(order1, order2, order3)
+
+        val wsc = mkWsInternalConnection()
+
+        orders.foreach(dex1.api.place)
+        orders.foreach(dex1.api.waitForOrderStatus(_, OrderStatus.Filled))
+
+        val buffer      = wsc.receiveAtLeastN[WsOrdersUpdate](1)
+        val orderEvents = buffer.orderEvents
+        orderEvents.keySet should matchTo(orders.map(_.id()).toSet)
+
+        orderEvents(order1.id()) should matchTo {
+          List(
+            mkExecutedFullOrder(
+              order1,
+              OrderStatus.Filled,
+              filledAmount = 2,
+              filledFee = 0.004,
+              avgWeighedPrice = 3,
+              executedAmount = 2,
+              executedFee = 0.004,
+              executedPrice = 3,
+              isMarket = false
+            ))
+        }
+
+        orderEvents(order2.id()) should matchTo {
+          List(
+            mkExecutedFullOrder(
+              order2,
+              OrderStatus.Filled,
+              filledAmount = 2,
+              filledFee = 0.003,
+              avgWeighedPrice = 2,
+              executedAmount = 2,
+              executedFee = 0.003,
+              executedPrice = 2,
+              isMarket = false
+            ))
+        }
+
+        orderEvents(order3.id()) should matchTo {
+          List(
+            mkExecutedFullOrder(
+              order3,
+              OrderStatus.Filled,
+              filledAmount = 4,
+              filledFee = 0.003,
+              avgWeighedPrice = 2.5,
+              executedAmount = 2,
+              executedFee = 0.0015,
+              executedPrice = 3,
+              isMarket = false
+            ),
+            mkExecutedFullOrder(
+              order3,
+              OrderStatus.PartiallyFilled,
+              filledAmount = 2,
+              filledFee = 0.0015,
+              avgWeighedPrice = 2,
+              executedAmount = 2,
+              executedFee = 0.0015,
+              executedPrice = 2,
+              isMarket = false
+            )
+          )
+        }
```

Comment:
Yeah, but it is comfortable for clients, because in most cases they need only the latest event. So they can process all messages from the first to the last and easily skip non interesting events.